### PR TITLE
fix(detail panel): Introduce error feedback to the user.

### PR DESCRIPTION
**Pull request by nimishamukherjee**
Introduce simple error feedback to the user. Currently users are clueless when they click on something and nothing happens on the screen. 
With this PR an error message will be displayed at the top of the detail panel (next to the work item type + work item ID).
If the backend supplies error status and status text, they will be displayed otherwise a default message, "Something went wrong. Try again." will be displayed.

### CODE REVIEW

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1974-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1974-fabric8-planner.int.rdu2c.fabric8.io

File: src/app/services/http-service.ts [hunk 1]
```diff
@@ -54,7 +54,7 @@ export class HttpService extends Http {
       console.log('retryWhen callback');
       let count = 0;
       return attempts.flatMap(error => {
-        if (error.status == 0) { // Server offline :: keep trying
+        if (error.status == 0 && count < 3) { // Server offline :: keep trying
```
Comment (sanbornsen) [inline src/app/services/http-service.ts]: error.status == 0 comes when internet is gone. We need to keep retrying when this happens. Not just thrice.

Comment (nimishamukherjee) [inline src/app/services/http-service.ts]: How long do we try? Would it make sense to display "No internet connection. Try after sometime." after 10 or 100 attempts.

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1974-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1974-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1974-fabric8-planner.int.rdu2c.fabric8.io

**Comment by fabric8cd** [comment]
@nimishamukherjee snapshot fabric8-ui is deployed and available for testing at https://fabric8-ui-fabric8-ui-pr-1974-fabric8-planner.int.rdu2c.fabric8.io

**Pull request merged**